src/Reflection/SignatureMap/functionMap.php

```diff
@@ -9731,7 +9731,7 @@
 'ReflectionFunction::getNumberOfParameters' => ['int'],
 'ReflectionFunction::getNumberOfRequiredParameters' => ['int'],
 'ReflectionFunction::getParameters' => ['array<int,ReflectionParameter>'],
-'ReflectionFunction::getReturnType' => ['?ReflectionType'],
+'ReflectionFunction::getReturnType' => ['?ReflectionNamedType'],
```

Comment:
Hi, thanks for the PR! I disagree with the changes about ReflectionNamedType. I actually went the other way in 0.12.0: https://github.com/phpstan/phpstan-src/commit/b447242cff29f94cd425d79bca92eb6b8b8861bb

ReflectionNamedType is the only ReflectionType implementation in PHP 7.x, but that will stop being true in PHP 8 where there will also be ReflectionUnionType without the `getName()` method. So code that's relying on ReflectionNamedType will be broken on PHP 8. I recommend doing something like this instead in your code:

$returnType = $reflectionFunction->getReturnType();
if (!$returnType instanceof ReflectionNamedType) {
  throw new \Exception('PHP 8 is not yet supported.');
}
```

Or maybe even code the support for union types today! :)

Otherwise the PR is fine.

Reply:
I see. So I'll turn it around and I'll move getName() and isBuiltin() from ReflectionType to ReflectionNamedType.